sdk/core/amqp-common/src/auth/aad.ts

```diff
@@ -59,7 +59,9 @@ export class AadTokenProvider implements TokenProvider {
       );
     }
     if (credentials instanceof MSITokenCredentials) {
-      credentials.resource = Constants.aadEventHubsAudience;
+      if (!credentials.resource) {
```

Comment:
Why multiple `if`,  Can we merge these two `if` conditions?

Reply:
The first if is for credentials and not for resource 
(Even if it were, it would only cover type checks such as for undefined, but not for empty strings, etc that `!` helps with).

Reply:
Ah you mean just merging into one statement, yeah we can